nshmdb/query.py

```diff
@@ -0,0 +1,336 @@
+"""
```

Comment:
Nice work! Did you consider existing packages for this?

Reply:
Tried looking but couldn't find anything. There are really big expression parsing libraries (as in, made to build compilers with) that I could find but nothing as simple as "parse something that looks like mathematics and give me a tree".